src/state_machine.zig

```diff
@@ -2850,9 +2757,6 @@ pub fn StateMachineType(
             if (t2.flags.post_pending_transfer) {
```

Comment:
In the next `if (t2.flags.void_pending_transfer) {`, could you add `assert(t2.amount == p.amount)`?